internal/pkg/cli/pipeline_init.go

```diff
@@ -523,24 +419,149 @@ For more information, please refer to: https://git.io/JfDFD.`,
 	}
-	if len(envs) == 0 {
-		return nil, errNoEnvsInApp
+
+	var manifestExists bool
+	manifestPath, err := o.workspace.WritePipelineManifest(manifest)
+	if err != nil {
+		e, ok := err.(*workspace.ErrFileExists)
+		if !ok {
+			return fmt.Errorf("write pipeline manifest to workspace: %w", err)
+		}
+		manifestExists = true
+		manifestPath = e.FileName
 	}
-	return envs, nil
+
+	manifestPath, err = relPath(manifestPath)
+	if err != nil {
+		return err
+	}
+
+	manifestMsgFmt := "Wrote the pipeline manifest for %s at '%s'\n"
+	if manifestExists {
+		manifestMsgFmt = "Pipeline manifest file for %s already exists at %s, skipping writing it.\n"
+	}
+	log.Successf(manifestMsgFmt, color.HighlightUserInput(o.githubRepo), color.HighlightResource(manifestPath))
+	log.Infoln("The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.")
+	return nil
+}
+
+func (o *initPipelineOpts) createBuildspec() error {
+	artifactBuckets, err := o.artifactBuckets()
+	if err != nil {
+		return err
+	}
+	content, err := o.parser.Parse(buildspecTemplatePath, struct {
+		BinaryS3BucketPath string
+		Version            string
+		ArtifactBuckets    []artifactBucket
+	}{
+		BinaryS3BucketPath: binaryS3BucketPath,
+		Version:            version.Version,
+		ArtifactBuckets:    artifactBuckets,
+	})
+	if err != nil {
+		return err
+	}
+	buildspecPath, err := o.workspace.WritePipelineBuildspec(content)
+	var buildspecExists bool
+	if err != nil {
+		e, ok := err.(*workspace.ErrFileExists)
+		if !ok {
+			return fmt.Errorf("write buildspec to workspace: %w", err)
+		}
+		buildspecExists = true
+		buildspecPath = e.FileName
+	}
+	buildspecMsgFmt := "Wrote the buildspec for the pipeline's build stage at '%s'\n"
+	if buildspecExists {
+		buildspecMsgFmt = "Buildspec file for pipeline already exists at %s, skipping writing it.\n"
+	}
+	buildspecPath, err = relPath(buildspecPath)
+	if err != nil {
+		return err
+	}
+	log.Successf(buildspecMsgFmt, color.HighlightResource(buildspecPath))
+	log.Infoln("The buildspec contains the commands to build and push your container images to your ECR repositories.")
+
+	return nil
+}
+
+func (o *initPipelineOpts) createSecretName() string {
+	return fmt.Sprintf("github-token-%s-%s", o.appName, o.githubRepo)
+}
+
+func (o *initPipelineOpts) createPipelineName() string {
+	return fmt.Sprintf("pipeline-%s-%s-%s", o.appName, o.githubOwner, o.githubRepo)
+}
+
+func (o *initPipelineOpts) createPipelineProvider() (manifest.Provider, error) {
+	config := &manifest.GitHubProperties{
+		OwnerAndRepository:    "https://" + githubURL + "/" + o.githubOwner + "/" + o.githubRepo,
+		Branch:                o.gitBranch,
+		GithubSecretIdKeyName: o.secretName,
+	}
+	return manifest.NewProvider(config)
+}
+
+func (o *initPipelineOpts) getEnvConfig(environmentName string) (*config.Environment, error) {
+	for _, env := range o.envs {
+		if env.Name == environmentName {
+			return env, nil
+		}
+	}
```

Comment:
After refactoring the env select logic to the `selector` pkg, we probably will ditch `o.envs` and need to call `GetEnvironment` for each of the environment here.

Reply:
Or does it make sense to keep an array of EnvConfigs for the environments in o.environment?

Reply:
hmm i thought `o.environments` can't be env configs because we use it as a variable to accept the flag values. we can figure it out what the best practice is in the next PR when the env refactoring happens!

Reply:
Oh, I meant keeping two arrays. Overkill? Maybe a time vs space consideration? Okay, yes, let's punt til next PR.

Reply:
Update: you're right-- calling `GetEnvironment` is the way to go.